orttraining/orttraining/core/graph/gradient_builder_base.cc

```diff
@@ -86,10 +86,10 @@ void ComputeBroadcastBackwardAxes(
 }
 
 std::vector<Dimension> GetShape(const ArgDef& arg_def) {
-  ORT_ENFORCE(arg_def.type_proto
-              && arg_def.type_proto->has_tensor_type()
-              && arg_def.type_proto->tensor_type().has_shape(),
-              "During GetShape, ", arg_def.name, "'s shape is null.");
+  //Throw OnnxruntimeShapeException type, so that it can specifically be caught in the gradient builder
```

Comment:
I don't think using exception is the right protocal here. It's OK to not have shape, we are just adding handling for this. 

Instead, use a more straightforward way to indicate the scuccess/failure of getting shape. 

bool GetShape(const ArgDef& arg_def, std::vecto&r<Dimension>) 
or 
Status GetShape(const ArgDef& arg_def, std::vecto&r<Dimension>) 

would just work fine here.